### DISCUSSION TITLE
Utilize RAPTOR on content retrieved?

### DISCUSSION THREAD

**Discussion by danieldekay** [assafelovic/gpt-researcher#780] [Ideas]
I found out about [RAPTOR](https://nextbrain.ai/blog/from-rag-to-raptor-improving-ai-retrieval-with-open-source-models#:~:text=RAPTOR%20takes%20AI%20retrieval%20a%20step%20further.%20Unlike,overcoming%20limitations%20of%20retrieving%20only%20short%20text%20chunks.) which promises better contextual performance compared to basic chunked RAG.

https://github.com/parthsarthi03/raptor

Has anyone tried this yet?

**Comment by a-yuksel** [comment]
Thanks for sharing, it seems like a prominent RAG improvement.

I am using a similar idea manually for questions where I have a rough idea about the potential answer. The process is like this:

- When I want to ask a question to an LLM, I also generate a short context for RAG. It is typically a candidate answer to that question.
- Then rather than using the embedding of the original question, I use the embedding of the (question + answer) in RAG
- If you are asking same/similar questions repetitively, you can use the answers in the previous steps to create a RAG context 
- This works well for me where the potential answers to the question can be identified beforehand, e.g. you have a question with a few potential answers, you have a rough idea about potential answers but don't know which one is the correct answer, plus don't know the details of the answer
- It will be restrictive and inefficient if the question is an open ended one. In that case, an LLM generated context like in RAPTOR can be more useful 

I am a new user of GPTResearcher. 
Does GPTResearcher allow adding a user-supplied context for RAG?

**Reply by ElishaKay** [reply]
Sup @a-yuksel 

Correct you can [set a DOC_PATH like so](https://github.com/assafelovic/gpt-researcher#-research-on-local-documents)

This doc_path will be leveraged when report_type equals `local` or `hybrid`

If you run with docker, these report types will also be available on localhost:3000

A PR was opened this morning to persist the vectors generated by these local docs across runs 🥳

**Reply by a-yuksel** [reply]
Thanks
